reconcile/external_resources/secrets_sync.py

```diff
@@ -99,54 +118,95 @@ def _init_ocmap(self, specs: Iterable[ExternalResourceSpec]) -> OCMap:
             integration=QONTRACT_INTEGRATION,
         )
 
-    @retry()
-    def _write_secrets_to_vault(self, spec: ExternalResourceSpec) -> None:
-        if spec.secret:
-            secret_path = f"{self.vault_path}/{QONTRACT_INTEGRATION}/{spec.cluster_name}/{spec.namespace_name}/{spec.output_resource_name}"
-            stringified_secret = {k: str(v) for k, v in spec.secret.items()}
-            desired_secret = {"path": secret_path, "data": stringified_secret}
-            self.vault_client.write(desired_secret, decode_base64=False)
+    def _copy_spec(self, spec: ExternalResourceSpec) -> ExternalResourceSpec:
+        copy = dataclasses.replace(spec)
+        copy.secret = dict(spec.secret)
+        return copy
```

Comment:
It does not work because `spec.secret` has an `Init=False` and the attribute can not be used in the constructor, hence not with `replace`.
This code no longer exists so not a problem anymore.